include/alpaka/dev/DevUniformCudaHipRt.hpp

```diff
@@ -171,30 +172,15 @@ namespace alpaka
 
         //! The CUDA/HIP RT preferred device warp size get trait specialization.
         template<typename TApi>
-        struct GetPreferredWarpSize<DevUniformCudaHipRt<TApi>>
+        struct GetPreferredWarpSize<DevUniformCudaHipRt<TApi>, std::enable_if_t<std::is_same_v<ApiCudaRt, TApi>>>
         {
-            ALPAKA_FN_HOST static auto getPreferredWarpSize(DevUniformCudaHipRt<TApi> const& dev) -> std::size_t
+            ALPAKA_FN_HOST static auto getPreferredWarpSize(DevUniformCudaHipRt<TApi> const& /* dev */) -> std::size_t
             {
-                typename TApi::DeviceProp_t devProp;
-                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(TApi::getDeviceProperties(&devProp, dev.getNativeHandle()));
-
-                return static_cast<std::size_t>(devProp.warpSize);
-            }
-        };
-
-#    ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
-        //! The CUDA RT preferred device warp size get trait specialization.
-        template<>
-        struct GetPreferredWarpSize<DevUniformCudaHipRt<ApiCudaRt>>
-        {
-            ALPAKA_FN_HOST static constexpr auto getPreferredWarpSize(DevUniformCudaHipRt<ApiCudaRt> const& /* dev */)
-                -> std::size_t
-            {
-                // All CUDA GPUs to date have a warp size of 32 threads.
-                return 32u;
+                // Same value for all cuda architectures
+                constexpr size_t warpSize{32};
+                return warpSize;
```

Comment:
please revert this to
```suggestion
                // All CUDA GPUs to date have a warp size of 32 threads.
                return 32u;
```

Reply:
@mehmetyusufoglu Could you please address both suggestions from @fwyzard